### ISSUE TITLE
关于损失函数和公式(3)-(5)的疑问

### ISSUE THREAD

**Issue by NanCheng2001** [fcyycf/RobustSplat#2]
您好，感谢您伟大的工作。我想问一下，论文中的公式（4）是否应该改成1-M_cos。
`where M_coswill be 1 is the feature cosine similarity is 1, and it will be 0 if the similarity is less than 0.5. Then the feature robust loss is expressed as:`
上述论文中的表述是否有错误。在我看来，feature cosine similarity越趋近于1的话，应当表示渲染特征与GT特征很相似，即越不可能是瞬态物体区域，所以M_cos应当趋近于0而不是1。（如果您的掩码将1当做瞬态物体区域的话）

请问我的理解有错吗？希望得到进一步的讨论和指正，感激不尽：
相应的，train.py的代码也应当修改：

```
            cosine = (1.-F.cosine_similarity(gt_feature, render_feature, dim=0).unsqueeze(0).sub(0.5).div(0.5)).clip(0.,1.)
            cosine = 1. - interpolation(cosine, image.shape[1], image.shape[2])
```

**Comment by fcyycf** [comment]
您好！感谢您的关注！
事实上，掩码为0的被认为是瞬态物体区域
您可以通过对应部分的代码快速理解：
```bash
# train.py, L156
Ll1 = (loss_mask * torch.abs((image - gt_image))).mean()
```
```bash
# train.py, L196-L197
cosine = (1.-F.cosine_similarity(gt_feature, render_feature, dim=0).unsqueeze(0).sub(0.5).div(0.5)).clip(0.,1.)
cosine = 1. - interpolation(cosine, image.shape[1], image.shape[2])

# train.py, L203 
mask_loss = 0.5 * torch.abs(mask - cosine).mean() + 0.5 * residual_loss + reg_loss 
```
